packages/instrumentation-browser-navigation/src/instrumentation.ts

```diff
@@ -0,0 +1,398 @@
+  private _onLoadHandler?: () => void;
+  private _onPopStateHandler?: (ev: PopStateEvent) => void;
+  private _onNavigateHandler?: (ev: Event) => void;
+
+  /**
+   *
+   * @param config
+   */
+  constructor(config: BrowserNavigationInstrumentationConfig) {
+    super(PACKAGE_NAME, PACKAGE_VERSION, config);
+    this.eventLogger = logs.getLogger(PACKAGE_NAME, PACKAGE_VERSION);
+    this.applyCustomLogRecordData = config?.applyCustomLogRecordData;
+  }
+
+  init() {}
+
+  /**
+   * callback to be executed when using hard navigation
+   */
+  private _onHardNavigation() {
+    const navLogRecord: LogRecord = {
+      eventName: EVENT_NAME,
+      attributes: {
+        [URL_FULL_ATTRIBUTE]: this._sanitizeUrl(document.documentURI),
+        [NAV_SAME_DOCUMENT]: false,
+        [NAV_HASH_CHANGE]: false,
+      },
+    };
+    this._applyCustomLogRecordData(navLogRecord, this.applyCustomLogRecordData);
+    this.eventLogger?.emit(navLogRecord);
+  }
+
+  /**
+   * callback to be executed when using soft navigation
+   */
+  private _onSoftNavigation(changeState: string | null | undefined, navigationEvent?: any) {
+    const referrer = this.oldUrl;
+    const currentUrl = (changeState === 'navigate' && navigationEvent?.destination?.url) 
+      ? navigationEvent.destination.url 
+      : location.href;
+    
+    if (referrer === currentUrl) {
+      return;
+    }
+    
+    const navType = this._mapChangeStateToType(changeState, navigationEvent);
+    const sameDocument = this._determineSameDocument(changeState, navigationEvent, referrer, currentUrl);
+    const hashChange = this._determineHashChange(changeState, navigationEvent, referrer, currentUrl);
+    const navLogRecord: LogRecord = {
+      eventName: EVENT_NAME,
+      attributes: {
+        [URL_FULL_ATTRIBUTE]: this._sanitizeUrl(currentUrl),
+        [NAV_SAME_DOCUMENT]: sameDocument,
+        [NAV_HASH_CHANGE]: hashChange,
+        ...(navType ? { [NAV_TYPE]: navType } : {}),
+      },
+    };
+    this._applyCustomLogRecordData(navLogRecord, this.applyCustomLogRecordData);
+    this.eventLogger?.emit(navLogRecord);
+    
+    this.oldUrl = currentUrl;
+  }
+
+  public _setLogger(eventLogger: Logger) {
+    this.eventLogger = eventLogger;
+  }
+
+  /**
+   * executes callback {_onDOMContentLoaded } when the page is viewed
+   */
+  private _waitForPageLoad() {
+    // Ensure previous handler is removed before adding a new one
+    if (this._onLoadHandler) {
+      document.removeEventListener('DOMContentLoaded', this._onLoadHandler);
+    }
+    this._onLoadHandler = this._onHardNavigation.bind(this);
+    document.addEventListener('DOMContentLoaded', this._onLoadHandler);
+  }
+
+  /**
+   * implements enable function
+   */
+  override enable() {
+    const cfg = this.getConfig() as BrowserNavigationInstrumentationConfig;
+    const useNavigationApiIfAvailable = !!cfg.useNavigationApiIfAvailable;
+    const hasNavigationApi = useNavigationApiIfAvailable && (window as any).navigation;
```

Comment:
nit: this expression evaluates to `false` or a reference the the `navigation` API. Why not renaming it to `navigationApi` and reuse it below. Then you could skip some extra checks like the one in L138 and use it directly in L146 instead of casting again the window object.

bonus: Since we are only using the `EventTarget` methods to add/remove event listeners we could to the following to get better types

```suggestion
    const navigationApi = useNavigationApiIfAvailable && (window as any).navigation as EventTarget;
```